src/middlewared/middlewared/plugins/pool.py

```diff
@@ -3101,6 +3101,15 @@ async def __common_validation(self, verrors, schema, data, mode):
             for i in ('force_size', 'sparse', 'volsize', 'volblocksize'):
                 if i in data:
                     verrors.add(f'{schema}.{i}', 'This field is not valid for FILESYSTEM')
+
+            if 'special_small_block_size' in data and not (
+                data['special_small_block_size'] == 0 or 512 <= data['special_small_block_size'] <= 1000000
+            ):
+                verrors.add(
+                    f'{schema}.special_small_block_size',
+                    'Valid values for this field are 0 or an integer value above '
```

Comment:
Sorry to be OCD, but there is 1 last minor issue with the validation text. Using the correct mathematical grammar will help clear this up.

`Valid values for this field are 0 or greater than or equal to 512 or less than or equal to 100000`